### PR TITLE
Refine share extension voice over

### PR DESCRIPTION
**Pull request by szabolcs-instructure**
refs: MBL-16342
affects: Student
release note: Refined share extension VoiceOver 

test plan: See ticket

## Screenshots

<table>
<tr><th>Voice Over video</th></tr>
<tr>
<td><video src="https://user-images.githubusercontent.com/110813321/202455564-d146f2d8-f411-4775-9ffc-6f2c42270f12.mov" maxHeight=500></td>
</tr>
</table>

## Checklist

- [ ] Follow-up e2e test ticket created or not needed
- [x] A11y checked
- [x] Tested on phone
- [ ] Tested on tablet
- [ ] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  One or more files are below the minimum test coverage 50%
  </td>
    </tr>
  </tbody>
</table>


Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.73% | 91.73% | 0%
Core/Core/Submissions/SubmissionBreakdown/SubmissionBreakdownViewModelProtocol.swift | 0% | 0% | 0%
Core/Core/Quizzes/QuizDetails/ViewModel/PreviewQuizDetailsViewModel.swift | 0% | 0% | 0%

[MBL-16342](https://instructure.atlassian.net/browse/MBL-16342)

#### Affected Apps: Student

#### Release Note: 
Refined share extension VoiceOver
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/SubmitAssignmentExtension/ViewModel/CoursePickerViewModel.swift [hunk 1]
```diff
@@ -16,13 +16,16 @@
...
+import Combine
 import SwiftUI
 
 public class CoursePickerViewModel: ObservableObject {
     public typealias Course = IdentifiableName
 
     @Published public var selectedCourse: CoursePickerViewModel.Course?
     @Published public var state: ViewModelState<[Course]> = .loading
+    public private(set) lazy var dismissView: AnyPublisher<Void, Never> = dismissViewSubject.eraseToAnyPublisher()
+    private let dismissViewSubject = PassthroughSubject<Void, Never>()
```
Comment (vargaat) [inline Core/Core/SubmitAssignmentExtension/ViewModel/CoursePickerViewModel.swift:28]: For simplicity's sake, let's expose the `PassthroughSubject` to have less boilerplate.

**Review by vargaat — CHANGES_REQUESTED** [review]

File: Core/Core/SubmitAssignmentExtension/View/SubmitAssignmentExtensionView.swift [hunk 2]
```diff
@@ -16,9 +16,215 @@
...
+    }
+
+    private func showFileProgressView(_ viewModel: FileProgressListViewModel) {
+        let listView = FileProgressListView(viewModel: viewModel)
+        let listViewController = CoreHostingController(listView)
+        env.router.show(listViewController, from: viewController, options: .modal(isDismissable: false, embedInNav: true, addDoneButton: false), analyticsRoute: "/file_progress")
+    }
+}
+
+public struct IOS14SubmitAssignmentExtensionView: View {
```
[the remark this answers is not in the document]
Comment (szabolcs-instructure) [reply Core/Core/SubmitAssignmentExtension/View/SubmitAssignmentExtensionView.swift:227]: Added deprecated for 14 and will be obsolete when we bump the minimum target for 15

File: Core/Core/SubmitAssignmentExtension/ViewModel/CoursePickerViewModel.swift (hunk 1 above)
Comment (szabolcs-instructure) [reply Core/Core/SubmitAssignmentExtension/ViewModel/CoursePickerViewModel.swift:28]: Fixed!

**Review by vargaat — APPROVED** [review]
QA+1

**Review by idevainstr — APPROVED** [review]
QA +1

**Pull request merged**